File: backend/app/Models/Korong.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Korong extends Model
{
    protected $fillable = [
        'name', 'address', 'phone', 'email', 'description', 'wali_korong_id', 'nagari_id'
    ];

    public function waliKorong(): BelongsTo
    {
        return $this->belongsTo(WaliKorong::class, 'wali_korong_id');
    }

    public function nagari(): BelongsTo
    {
        return $this->belongsTo(Nagari::class, 'nagari_id');
    }
}

File: kkn_backend/app/models/korong.go
package models

import (
	"github.com/goravel/framework/database/orm"
	"github.com/goravel/framework/facades"
)

type Korong struct {
	orm.Model

	Name         string
	Address      string
	Phone        string
	Email        string
	Description  string
	WaliKorongId uint64

	WaliKorong *WaliKorong

	Coordinates []Coordinate `orm:"morphMany:Coordinateable"`
	Images      []Image      `orm:"morphMany:Imageable"`
}

func (korong *Korong) Create() error {
	return facades.Orm().Query().Create(korong)
}
